src/Symfony/Component/VarExporter/Tests/LazyGhostTraitTest.php

```diff
@@ -330,4 +331,56 @@ public function testReflectionPropertyGetValue()
 
         $this->assertSame(-3, $r->getValue($obj));
     }
+
+    public function testFullPartialInitialization()
+    {
+        $counter = 0;
+        $initializer = static function (ChildTestClass $instance, string $property, ?string $scope, mixed $default) use (&$counter) {
+            return 234;
+        };
+        $instance = ChildTestClass::createLazyGhost([
+            'public' => $initializer,
+            'publicReadonly' => $initializer,
+            "\0*\0protected" => $initializer,
+            "\0" => function ($obj, $defaults) use (&$instance, &$counter) {
+                $counter += 1000;
+                $this->assertSame($instance, $obj);
+
+                return [
+                    'public' => 345,
+                    'publicReadonly' => 456,
+                    "\0*\0protected" => 567,
+                ] + $defaults;
+            },
+        ]);
+
+        $this->assertSame($instance, $instance->initializeLazyObject());
+        $this->assertSame(345, $instance->public);
+        $this->assertSame(456, $instance->publicReadonly);
+        $this->assertSame(6, ((array) $instance)["\0".ChildTestClass::class."\0private"]);
+        $this->assertSame(3, ((array) $instance)["\0".TestClass::class."\0private"]);
+        $this->assertSame(1000, $counter);
+    }
+
+    public function testPartialInitializationFallback()
+    {
+        $counter = 0;
+        $instance = ChildTestClass::createLazyGhost([
+            "\0" => function ($obj) use (&$instance, &$counter) {
+                $counter += 1000;
+                $this->assertSame($instance, $obj);
+
+                return [
+                    'public' => 345,
+                    'publicReadonly' => 456,
+                    "\0*\0protected" => 567,
+                ];
+            },
+        ], []);
+
+        $this->assertSame(345, $instance->public);
+        $this->assertSame(456, $instance->publicReadonly);
+        $this->assertSame(567, ((array) $instance)["\0*\0protected"]);
+        $this->assertSame(1000, $counter);
+    }
```

Comment:
I suggest adding another test case that tests full initialisation after partial initialisation to ensure that previously initialised properties are not overwritten:
```php
    public function testFullInitializationAfterPartialInitialization()
     {
         $counter = 0;
         $initializer = static function (ChildTestClass $instance, string $property, ?string $scope, mixed $default) use (&$counter) {
             return 234;
         };
         $instance = ChildTestClass::createLazyGhost([
             'public' => $initializer,
             'publicReadonly' => $initializer,
             "\0*\0protected" => $initializer,
             "\0" => function ($obj, $defaults) use (&$instance, &$counter) {
                 $counter += 1000;
                 $this->assertSame($instance, $obj);

                 return [
                     'public' => 345,
                     'publicReadonly' => 456,
                     "\0*\0protected" => 567,
                 ] + $defaults;
             },
         ]);

         $this->assertSame(234, $instance->public);

         $this->assertSame($instance, $instance->initializeLazyObject());
         $this->assertSame(234, $instance->public);
         $this->assertSame(456, $instance->publicReadonly);
         $this->assertSame(6, ((array) $instance)["\0".ChildTestClass::class."\0private"]);
         $this->assertSame(3, ((array) $instance)["\0".TestClass::class."\0private"]);
         $this->assertSame(1000, $counter);
     }
```

Reply:
test added, thanks for providing it!